numpy/core/tests/test_deprecations.py

```diff
@@ -37,119 +38,17 @@ def setUp(self):
         # (While checking them set to "error" those are ignored anyway)
         # We still have them show up, because otherwise they would be raised
-        warnings.filterwarnings("always", category=np.VisibleDeprecationWarning)
+        warnings.filterwarnings("always", category=self.warning_cls)
         warnings.filterwarnings("always", message=self.message,
-                                category=np.VisibleDeprecationWarning)
+                                category=self.warning_cls)
 
     def tearDown(self):
         self.warn_ctx.__exit__()
 
     def assert_deprecated(self, function, num=1, ignore_others=False,
                           function_fails=False,
-                          exceptions=(np.VisibleDeprecationWarning,),
+                          exceptions=np._NoValue,
                           args=(), kwargs={}):
-        """Test if VisibleDeprecationWarnings are given and raised.
-
-        This first checks if the function when called gives `num`
-        VisibleDeprecationWarnings, after that it tries to raise these
-        VisibleDeprecationWarnings and compares them with `exceptions`.
-        The exceptions can be different for cases where this code path
-        is simply not anticipated and the exception is replaced.
-
-        Parameters
-        ----------
-        function : callable
-            The function to test
-        num : int
-            Number of VisibleDeprecationWarnings to expect. This should
-            normally be 1.
-        ignore_others : bool
-            Whether warnings of the wrong type should be ignored (note that
-            the message is not checked)
-        function_fails : bool
-            If the function would normally fail, setting this will check for
-            warnings inside a try/except block.
-        exceptions : Exception or tuple of Exceptions
-            Exception to expect when turning the warnings into an error.
-            The default checks for DeprecationWarnings. If exceptions is
-            empty the function is expected to run successfully.
-        args : tuple
-            Arguments for `function`
-        kwargs : dict
-            Keyword arguments for `function`
-        """
-        # reset the log
-        self.log[:] = []
-
-        try:
-            function(*args, **kwargs)
-        except (Exception if function_fails else tuple()):
-            pass
-
-        # just in case, clear the registry
-        num_found = 0
-        for warning in self.log:
-            if warning.category is np.VisibleDeprecationWarning:
-                num_found += 1
-            elif not ignore_others:
-                raise AssertionError(
-                        "expected DeprecationWarning but got: %s" %
-                        (warning.category,))
-        if num is not None and num_found != num:
-            msg = "%i warnings found but %i expected." % (len(self.log), num)
-            lst = [w.category for w in self.log]
```

Comment:
This line was a bug, that was fixed in `_DeprecationTestCase` but not in this one